crates/services/upgradable-executor/src/executor.rs

```diff
@@ -517,19 +538,16 @@ where
             gas_price,
         };
 
-        let previous_block_height = if !dry_run {
-            block.header_to_produce.height().pred()
-        } else {
-            // TODO: https://github.com/FuelLabs/fuel-core/issues/2062
-            None
+        let db_height = match mode {
+            ProduceBlockMode::Produce => block.header_to_produce.height().pred(),
+            ProduceBlockMode::DryRunLatest => None,
+            ProduceBlockMode::DryRunAt { height } => height.pred(),
         };
```

Comment:
We shouldn't allow dry run in the past in the case of `InMemory` storage. It will create issues for local development. You can see why it was disabled:

https://github.com/FuelLabs/fuel-core/pull/2063